paddle/fluid/operators/uniform_random_op.h

```diff
@@ -98,5 +100,33 @@ inline std::vector<int64_t> GetNewDataFromShapeTensorList(
   return vec_new_shape;
 }
 
+template <typename T>
+inline void UniformRealDistribution(T* data, const int64_t& size,
+                                    const float& min, const float& max,
+                                    const unsigned int& seed) {
+  VLOG(4) << "[CPU] UniformRandomKernel<T>";
+  std::uniform_real_distribution<T> dist(static_cast<T>(min),
+                                         static_cast<T>(max));
+  auto engine = framework::GetCPURandomEngine(seed);
+
+  for (int64_t i = 0; i < size; ++i) {
+    data[i] = dist(*engine);
+  }
+}
+
+template <>
+inline void UniformRealDistribution(platform::bfloat16* data,
+                                    const int64_t& size, const float& min,
+                                    const float& max,
+                                    const unsigned int& seed) {
+  VLOG(4) << "[CPU] UniformRandomKernel<bfloat16>";
+  std::uniform_real_distribution<float> dist(min, max);
+  auto engine = framework::GetCPURandomEngine(seed);
+
+  for (int64_t i = 0; i < size; ++i) {
+    data[i] = static_cast<platform::bfloat16>(dist(*engine));
+  }
+}
+
```

Comment:
Those two function templates are an implementation detail, thus should not be placed in header file. Please move them into the source file and put into anonymous namespace or make them static at global namespace to prevent cluttering global/local namespaces.

Reply:
Yes, you are right it will be a better solution.